src/clients/storage/StorageClientBase.h

```diff
@@ -47,46 +48,67 @@ class StorageRpcResponse final {
     return maxLatency_;
   }
 
-  void setLatency(HostAddr host, int32_t latency, int32_t e2eLatency) {
-    std::lock_guard<std::mutex> g(*lock_);
+  void setLatencyUnsafe(const HostAddr& host, int32_t latency, int32_t e2eLatency) {
     if (latency > maxLatency_) {
       maxLatency_ = latency;
     }
     hostLatency_.emplace_back(std::make_tuple(host, latency, e2eLatency));
   }
 
-  void markFailure() {
+  void setLatency(const HostAddr& host, int32_t latency, int32_t e2eLatency) {
```

Comment:
Seems there no usage of the previouse safe version? if we only call these functions with single thread, just delete the lock.

Reply:
yeah. There's no usage for these functions in multiple thread scenario. I'll delete the lock from the class.